packages/core/src/modules/dids/methods/peer/PeerDidRegistrar.ts

```diff
@@ -37,23 +37,27 @@ export class PeerDidRegistrar implements DidRegistrar {
         const seed = options.secret?.seed
         const privateKey = options.secret?.privateKey
 
-        if (!keyType) {
+        let key = options.options.key
+
+        if (!key && keyType) {
+          key = await agentContext.wallet.createKey({
+            keyType,
+            seed,
+            privateKey,
+          })
+        }
+
+        if (!key) {
```

Comment:
need to handle where both keyType and key are passed, but they don't match

Reply:
also need to handle the case where privateKey/seed is passed and key as well. That should cause an error. As it results in ambigious behaviour (I have no idea without looking at the code which will take precedence)

Reply:
(same for all files)